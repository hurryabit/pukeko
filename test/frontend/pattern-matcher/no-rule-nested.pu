data List a = Nil | Cons a (List a)
data Option a = None | Some a
f :: a -> List (Option a) -> a
f x0 ys =
  case ys of
    Nil              -> x0
    Cons (Some x) _  -> x
    Cons _        ys -> f x0 ys
