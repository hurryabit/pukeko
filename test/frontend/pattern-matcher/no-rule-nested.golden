test/frontend/pattern-matcher/no-rule-nested.pu:9:3: cannot apply constructor rule
